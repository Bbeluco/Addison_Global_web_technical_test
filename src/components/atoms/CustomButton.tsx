import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';

type CustomButtonType = {
  text: string;
  styleButton: StyleSheet.NamedStyles<any> | StyleSheet.NamedStyles<unknown>;
  styleTextButton:
    | StyleSheet.NamedStyles<any>
    | StyleSheet.NamedStyles<unknown>;
};

export function CustomButton({
  text,
  styleButton,
  styleTextButton,
}: CustomButtonType): React.JSX.Element {
  return (
    <TouchableOpacity style={styleButton}>
      <Text style={styleTextButton}>{text}</Text>
    </TouchableOpacity>
  );
}
