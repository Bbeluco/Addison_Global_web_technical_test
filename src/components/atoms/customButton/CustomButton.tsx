import React from 'react';
import {TouchableOpacity, Text} from 'react-native';
import {CustomButtonType} from '@customTypes/customButtonType';

export function CustomButton({
  text,
  styleParent,
  styleChildren,
}: CustomButtonType): React.JSX.Element {
  return (
    <TouchableOpacity style={styleParent}>
      <Text style={styleChildren}>{text}</Text>
    </TouchableOpacity>
  );
}
