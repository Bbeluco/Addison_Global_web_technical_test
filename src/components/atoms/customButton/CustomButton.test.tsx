import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {CustomButton} from './CustomButton';

describe('<CustomButton />', () => {
  it('should render and reflect the text passed as props', () => {
    const text = 'Unit test';
    render(<CustomButton styleChildren={{}} styleParent={{}} text={text} />);
    expect(screen.queryByText(text)).not.toBeNull();
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
