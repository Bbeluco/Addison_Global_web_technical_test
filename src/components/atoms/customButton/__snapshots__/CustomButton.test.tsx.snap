// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> should render and reflect the text passed as props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={{}}
  >
    Unit test
  </Text>
</View>
`;
